core/lib/network-recorder.js

```diff
@@ -248,7 +248,8 @@ class NetworkRecorder extends RequestEventEmitter {
 
     let candidates = recordsByURL.get(initiatorURL) || [];
     // The initiator must come before the initiated request.
-    candidates = candidates.filter(cand => cand.responseReceivedTime <= record.startTime);
+    candidates = candidates
+      .filter(cand => cand.responseHeadersEndTime <= record.networkRequestTime);
```

Comment:
i think this could be rendererStartTime instead of networkRequestTime.
as we have to go back to renderer to make new requests. 
redirect would be the only case where that wouldnt happen but 1) thats cuz we artificially separate them into reqs and 2) i don't think we'll hit this situation with redirects.